scripts/HPC_chatbot_preprocessor/chatbot_parser.py

```diff
@@ -0,0 +1,683 @@
+
+    # Read the YAML file
+    with open(yml_file_path, 'r') as yml_file:
+        words_dict = yaml.safe_load(yml_file)
+
+    # ugly fix for index.md error that occurs because of the macro "config.repo_url" in mkdocs/docs/HPC/index.md
+    additional_context = {
+        'config': {
+            'repo_url': REPO_URL
+        }
+    }
+    combined_context = {**words_dict, **additional_context}
+
+    # Mangle the OS-related if-statements
+    mangle_ifs(copy_location, filename)
+
+    # Use Jinja2 to replace the macros
+    template_loader = ChoiceLoader([FileSystemLoader(searchpath=IF_MANGLED_FILES), FileSystemLoader(searchpath=os.path.join(RETURN_DIR, RETURN_DIR, MKDOCS_DIR, DOCS_DIR, HPC_DIR))])
+    templateEnv = Environment(loader=template_loader)
+    template = templateEnv.get_template(filename)
+    rendered_content = template.render(combined_context)
+
+    # Save the rendered content to a new file
+    with open(copy_location, 'w', encoding='utf-8', errors='ignore') as output_file:
+        output_file.write(rendered_content)
+
+
+def mangle_os_ifs(line, is_os):
+    """
+    function that mangles the os-related if-statements. This is needed because we want to keep these if-statements intact after jinja-parsing to build the directory structure.
+    We don't want to mangle all if-related statements (such as else and endif) so we need to keep track of the context of the last few if-statements.
+
+    :param line: the current line to check for os-related if-statements
+    :param is_os: variable keep track of the current os-state of the if-statements. Can be NON_OS_IF, NON_OS_IF_IN_OS_IF, OS_IF or OS_IF_IN_OS_IF 
+        NON_OS_IF: not in an os-if
+        NON_OS_IF_IN_OS_IF: in a non-os-if nested in an os-if
+        OS_IF: in an os-if
+        OS_IF_IN_OS_IF: in an os-if nested in an os-if
+    :return line: the modified line with  mangled os-related if-statements
+    """
+
+    match = re.search(r'\{%(.*?)%}(.*)', line)
+
+    start_index = 0
+    added_length = 0
+
+    while match:
+
+        constr_match = re.search(r'\{%.*?%}', match.string)
+        if_match = re.search(r'if ', match.group(1))
+        if_os_match = re.search(r'if OS ', match.group(1))
+        endif_match = re.search(r'endif', match.group(1))
+
+        # mangle positions
+        pos_first_mangle = constr_match.start() + start_index + added_length + 1
+        pos_second_mangle = constr_match.end() + start_index + added_length - 1
+
+        # different parts of the original string
+        PART_BEFORE_MANGLING = line[:pos_first_mangle]
+        PART_BETWEEN_MANGLING = line[pos_first_mangle:pos_second_mangle]
+        PART_AFTER_MANGLING = line[pos_second_mangle:]
+
+        # this logic isn't flawless, there are number of nested if-constructions that are technically possible that would break this logic, but these don't appear in the documentation as it doesn't make sense to have these
```

Comment:
can you add what you support in the readme? so we at least know what not to try?